app/src/main/java/com/nmakademija/nmaakademija/adapter/ScheduleAdapter.java

```diff
@@ -1,69 +1,74 @@
         }
-        return convertView;
     }
 
-    private View getViewScheduleEvent (ScheduleEvent scheduleEvent, View convertView, ViewGroup parent){
-        convertView = LayoutInflater.from(getContext()).inflate(R.layout.schedule_event, parent, false);
+    private Context context;
+    private List<ScheduleItem> events;
 
-        TextView scheduleStartTime = (TextView) convertView.findViewById(R.id.schedule_start_time);
-        TextView scheduleEndTime = (TextView) convertView.findViewById(R.id.schedule_end_time);
-        TextView scheduleName = (TextView) convertView.findViewById(R.id.schedule_name);
-        TextView scheduleLecturers = (TextView) convertView.findViewById(R.id.schedule_author);
+    public ScheduleAdapter(Context context, List<ScheduleItem> events) {
+        this.events = events;
+        this.context = context;
 
-        scheduleStartTime.setText(scheduleEvent.getStartTime());
-        scheduleEndTime.setText(scheduleEvent.getEndTime());
-        scheduleName.setText(scheduleEvent.getName());
-        scheduleLecturers.setText(scheduleEvent.getLecturerName());
-
-        return convertView;
     }
 
-    private View getViewScheduleDayBanner (ScheduleDayBanner scheduleDayBanner, View convertView, ViewGroup parent){
-        convertView = LayoutInflater.from(getContext()).inflate(R.layout.schedule_day, parent, false);
-
-        TextView scheduleTime = (TextView) convertView.findViewById(R.id.schedule_time);
+    @Override
+    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.schedule_event, parent, false);
+        return new MyViewHolder(itemView);
+    }
 
-        scheduleTime.setText(scheduleDayBanner.getTime());
+    @Override
+    public void onBindViewHolder(MyViewHolder holder, int position) {
+        final ScheduleItem scheduleItem = events.get(position);
+        if (scheduleItem instanceof ScheduleEvent) {
+            ScheduleEvent scheduleEvent = (ScheduleEvent) scheduleItem;
+            holder.startTime.setText(scheduleEvent.getStartTime());
+            holder.endTime.setText(scheduleEvent.getEndTime());
+            holder.author.setText(scheduleEvent.getLecturerName());
+            holder.name.setText(scheduleEvent.getName());
+            holder.endTime.setVisibility(View.VISIBLE);
+            holder.startTime.setVisibility(View.VISIBLE);
+            holder.author.setVisibility(View.VISIBLE);
+            holder.name.setTextAppearance(context, R.style.Schedule_item_name);
+        } else {
+            ScheduleDayBanner scheduleDayBanner = (ScheduleDayBanner) scheduleItem;
+            holder.name.setTextAppearance(context, R.style.Schedule_day);
+            holder.name.setText(scheduleDayBanner.getTime());
+            holder.endTime.setVisibility(View.GONE);
+            holder.startTime.setVisibility(View.GONE);
+            holder.author.setVisibility(View.GONE);
```

Comment:
Use two different `ViewHolders`. More about it here http://stackoverflow.com/questions/26245139/how-to-create-recyclerview-with-multiple-view-type